core/templates/pages/community-dashboard-page/services/contribution-opportunities-backend-api.service.ts

```diff
@@ -95,6 +103,33 @@ export class ContributionOpportunitiesBackendApiService {
     });
   }
 
+  private _fetchFeaturedTranslationLanguages(
+      successCallback: (
+        featuredTranslationLanguages: FeaturedTranslationLanguage[]
+      ) => void,
+      errorCallback: (reason: any) => void
+  ) {
+    this.http.get('/getfeaturedtranslationlanguages').toPromise()
+      .then((data: any) => {
+        const featuredTranslationLanguages = (
+          data.featured_translation_languages.map(
+            (backendDict: IFeaturedTranslationLanguageBackendDict) =>
+              this.readOnlyFeaturedTranslationLanguageObjectFactory
+                .createFromBackendDict(backendDict)
+          ));
+
+        if (successCallback) {
+          successCallback(featuredTranslationLanguages);
+        }
+      },
+      (error) => {
+        if (errorCallback) {
+          errorCallback(error);
+        }
+      });
```

Comment:
Is this necessary? Can't we just return `featuredTranslationLanguages` and `then()` will return a new Promise?

Reply:
Not entirely sure what you mean, but I did clean up the code here